src/cmd/create.go

```diff
@@ -401,7 +401,7 @@ func createContainer(container, image, release string, showCommandToEnter bool)
 	createArgs = append(createArgs, xdgRuntimeDirEnv...)
 
 	createArgs = append(createArgs, []string{
-		"--hostname", "toolbox",
+		"--hostname", container,
```

Comment:
I suspect that the rules for a valid hostname are different from the rules for a valid OCI container name.